app/src/main/java/com/google/maps/android/ktx/demo/MainActivity.kt

```diff
@@ -0,0 +1,236 @@
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+import com.google.maps.android.clustering.ClusterManager
+import com.google.maps.android.collections.GroundOverlayManager
+import com.google.maps.android.collections.MarkerManager
+import com.google.maps.android.collections.PolygonManager
+import com.google.maps.android.collections.PolylineManager
+import com.google.maps.android.data.Renderer.ImagesCache
+import com.google.maps.android.data.geojson.GeoJsonLineStringStyle
+import com.google.maps.android.data.geojson.GeoJsonPolygonStyle
+import com.google.maps.android.ktx.MapsExperimentalFeature
+import com.google.maps.android.ktx.awaitMap
+import com.google.maps.android.ktx.demo.io.MyItemReader
+import com.google.maps.android.ktx.demo.model.MyItem
+import com.google.maps.android.ktx.utils.geojson.geoJsonLayer
+import com.google.maps.android.ktx.utils.kml.kmlLayer
+import org.json.JSONException
+
+/**
+ * A demo of multiple layers on the map.
+ *
+ * To add your Maps API key to this project:
+ *   1. Create a file app/gradle.properties
+ *   2. Add this line, where YOUR_API_KEY is your API key:
+ *        MAPS_API_KEY="YOUR_API_KEY"
+ */
+class MainActivity : AppCompatActivity() {
+    var mIsRestore: Boolean = false
+
+    @MapsExperimentalFeature
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        mIsRestore = savedInstanceState != null
+        setContentView(R.layout.activity_main)
+        val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as? SupportMapFragment
+        lifecycle.coroutineScope.launchWhenCreated {
+            check(mapFragment != null)
+            val googleMap = mapFragment.awaitMap()
+            if (!mIsRestore) {
+                googleMap.moveCamera(
+                    CameraUpdateFactory.newLatLngZoom(
+                        LatLng(51.403186, -0.126446),
+                        10F
+                    )
+                )
+            }
+            showMapLayers(googleMap)
+        }
+    }
+
+    private fun showMapLayers(map: GoogleMap) {
+        // Shared object managers - used to support multiple layer types on the map simultaneously
+        val markerManager = MarkerManager(map)
+        val groundOverlayManager = GroundOverlayManager(map)
+        val polygonManager = PolygonManager(map)
+        val polylineManager = PolylineManager(map)
+
+        addClusters(map, markerManager)
+        addGeoJson(map, markerManager, polylineManager, polygonManager, groundOverlayManager)
+        addKml(map, markerManager, polylineManager, polygonManager, groundOverlayManager)
+        addMarker(markerManager)
+    }
+
+    private fun addClusters(map: GoogleMap, markerManager: MarkerManager) {
+        val clusterManager = ClusterManager<MyItem>(this, map, markerManager)
+        map.setOnCameraIdleListener(clusterManager)
```

Comment:
Should `ClusterManager` internally make this call? No changes needed here but I wonder if we should make that improvement in the AMU API if this should always be desired.

Reply:
Yeah, good question. I think the majority of use cases call this, although I could see certain use cases use static clusters that don't change on zoom levels. I'll open a separate issue to discuss on AMU repo.

Reply:
Issue opened at https://github.com/googlemaps/android-maps-utils/issues/681.